### PR TITLE
Support comparison of nested operator infos in TCK frame

### PR DESCRIPTION
**Pull request by Aiee**
As title.
Now cases like below are supported.
```
  Scenario: apply get nbrs, dedup and project merge opt rule
    When profiling query:
      """
      MATCH (v:player{name: 'Tim Duncan'})-[:like*0..1]->(v2)
      RETURN v2.name AS Name
      """
    Then the result should be, in any order:
      | Name            |
      | "Manu Ginobili" |
      | "Tony Parker"   |
      | "Tim Duncan"    |
    And the execution plan should be:
      | id | name               | dependencies | operator info                                       |
      | 0  | Project            | 1            |                                                     |
      | 1  | Filter             | 2            |                                                     |
      | 2  | Project            | 3            |                                                     |
      | 3  | InnerJoin          | 4            |                                                     |
      | 4  | Project            | 5            |                                                     |
      | 5  | GetVertices        | 6            | {"dedup": "true"}                                   |
      | 6  | Filter             | 7            |                                                     |
      | 7  | UnionAllVersionVar | 8            |                                                     |
      | 8  | Loop               | 15           | {"loopBody": "9"}                                   |
      | 9  | Filter             | 10           |                                                     |
      | 10 | Project            | 11           |                                                     |
      | 11 | InnerJoin          | 12           | {"inputVar": {"rightVar":{"__Project_11":"0"}}}     |
      | 12 | Project            | 13           |                                                     |
      | 13 | GetNeighbors       | 14           | {"dedup": "true"}                                   |
      | 14 | Start              |              |                                                     |
      | 15 | Project            | 16           |                                                     |
      | 16 | Filter             | 17           |                                                     |
      | 17 | GetVertices        | 18           | {"dedup": "true"}                                   |
      | 18 | IndexScan          | 19           | {"indexCtx": {"columnHints":{"scanType":"PREFIX"}}} |
      | 19 | Start              |              |                                                     |
```

Todos:
- [x] Fix TCK format of the multi-layer nested expected result
- [x] Support to compare using the expected result that is a subset of a nested Dict. e,g, No need to list every key/value of the inner Dict.
- [x] Support to specify id of plan node in plan table and check for dependencies

### CODE REVIEW

**Review by yixinglu — COMMENTED** [review]
Could you support to specify id of plan node in plan table, such as:

```
| id | name    | dependencies | operator info |
| 18 | Project | 17           |               |  
| 17 | Filter  | 16           |               |
```

File: tests/README.md [hunk 1]
```diff
@@ -134,7 +156,9 @@ Each feature file is composed of different scenarios which split test units into
 
 The table in `Then` step must have the first header line even if there's no data rows.
 
-`Background` is the common steps of different scenarios. Scenarios will be executed in parallel.
+`Background` is the common steps of different scenarios. Scenarios will be executed in parallel. 
+
+Note that for cases contain wxecutino plans, it is mandatory to fill the `id` column.
```
Comment (yixinglu) [inline tests/README.md:161]: fix typo: `execution`

Comment (Aiee) [reply tests/README.md]: Fixed

**Review by yixinglu — APPROVED** [review]

**Review by jievince — APPROVED** [review]
Great!

**Pull request merged**